invenio_records_lom/resources/serializers/schemas/ui.py

```diff
@@ -0,0 +1,251 @@
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Schemas for ui."""
+# TODO: remove the following lines after moving to python3.9
+# allow `list[dict]` annotation, works without future-import for python>=3.9
+from __future__ import annotations
+
+import re
+
+from flask import current_app
+from flask_resources import BaseObjectSchema
+from invenio_rdm_records.resources.serializers.ui.fields import AccessStatusField
+from invenio_rdm_records.resources.serializers.ui.schema import (
+    FormatDate,
+    record_version,
+)
+from marshmallow import fields
+from marshmallow_oneofschema import OneOfSchema
+from werkzeug.local import LocalProxy
+
+from ....services.schemas.fields import ControlledVocabularyField
+from .utils import get_newest_part, get_related, get_text
+
+
+class Title(fields.Field):
+    """Title Field."""
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        """Serialize."""
+        # TODO
+        # checkout how to get the actual preferred language and show that string if it exists!
+        return get_text(value)
+
+
+class Contributors(fields.Field):
+    """Contributors Field."""
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        """Serialize."""
+        contributors = []
+        for sub_obj in value:
+            contributors.append(
+                {
+                    "fullname": sub_obj["entity"],
+                    "role": get_text(sub_obj["role"]["value"]),
+                }
+            )
+        return contributors
+
+
+class GeneralDescriptions(fields.Field):
+    """General Descriptions Field."""
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        """Serialize."""
+        return list(map(get_text, value))
+
+
+class EducationalDescriptions(fields.Field):
+    """Educational Descriptions Field."""
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        """Serialize."""
+        return list(map(get_text, value))
+
+
+class Location(fields.Field):
+    """Location Field."""
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        """Serialize."""
+        return value["#text"]
+
+
+class Rights(fields.Field):
+    """Rights Field."""
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        """Serialize."""
+        right = {}
+        value = value or {}
+
+        if "creativecommons" in value.get("url", ""):
+            result = re.search("licenses/([a-z-]+)/.*", value["url"])
+            id_ = f"CC {result.group(1).upper()}"
+            icon = f"{id_.lower().replace(' ', '-')}-icon"
+            right = {
+                "id": id_,
+                "title_l10n": id_,
+                "description_l10n": id_,
+                "link": value["url"],
+                "icon": icon,
+            }
+
+        return [right]
+
+
+class LOMUIBaseSchema(BaseObjectSchema):
+    """Base schema for LOMUI-classes, containing all common fields."""
+
+    # publication_date_l10n_medium = FormatEDTF(...)
```

Comment:
is this code really necessary to keep it commented? if yes, please add a comment why it should stay. (applies to the whole commented block)

Reply:
whoops, definitely not necessary
this was for debugging landing page in last commit (it's all the fields that the comparable `invenio` template has)
removed